processing/src/main/java/org/apache/druid/query/rowsandcols/semantic/DefaultColumnSelectorFactoryMaker.java

```diff
@@ -173,7 +169,10 @@ private <T> T withColumnAccessor(String column, Function<ColumnAccessor, T> fn)
       ColumnAccessor retVal = accessorCache.get(column);
       if (retVal == null) {
         Column racColumn = rac.findColumn(column);
-        retVal = racColumn == null ? null : racColumn.toAccessor();
+        if (racColumn == null) {
+          throw DruidException.defensive("Column[%s] not found!", column);
+        }
+        retVal = racColumn.toAccessor();
```

Comment:
I don't think this is safe to commit.  Generally speaking, the contract around when we are asked for a column that doesn't exist is to pretend it does exist and is all `null`.  This is an important element of how we deal with schema-lessness and schema migration: fundamentally, there's no requirement that two segments have the exact same schema and so it's entirely possible and acceptable for a column that doesn't actually exist to be asked for.